apps/meteor/app/livechat/server/api/v1/visitor.ts

```diff
@@ -68,16 +68,41 @@ API.v1.addRoute('livechat/visitor', {
 			);
 		}
 
-		if (customFields && Array.isArray(customFields)) {
-			for await (const field of customFields) {
-				const customField = await LivechatCustomField.findOneById(field.key);
-				if (!customField) {
-					continue;
-				}
-				const { key, value, overwrite } = field;
-				if (customField.scope === 'visitor' && !(await VisitorsRaw.updateLivechatDataByToken(token, key, value, overwrite))) {
-					return API.v1.failure();
-				}
+		if (customFields && Array.isArray(customFields) && customFields.length > 0) {
+			const keys = customFields.map((field) => field.key);
+			const errors: string[] = [];
+
+			const processedKeys = await Promise.all(
+				await LivechatCustomField.findByIdsAndScope<Pick<ILivechatCustomField, '_id'>>(keys, 'visitor', {
+					projection: { _id: 1 },
+				})
+					.map(async (field) => {
+						const customField = customFields.find((f) => f.key === field._id);
+						if (!customField) {
+							return;
+						}
+
+						const { key, value, overwrite } = customField;
+						// TODO: Change this to Bulk update
+						if (!(await VisitorsRaw.updateLivechatDataByToken(token, key, value, overwrite))) {
```

Comment:
This is returning an `UpdateResult` from the DB, isn't it? If so, this will be always true.

Reply:
Yep, sadly. There's a case where it can return boolean | void iirc. But left this one exactly like this cause I plan to move this whole thing to a single batch update, so left the current behavior.

Reply:
I think this can't return `void` unless we provide a callback to the Mongo driver. If this is possible to return void somehow, please update that on the model method.

Reply:
Missed on the `void` part. Func returns `Document | Boolean | UpdateResult`

While the type assertions again could work, I plan to refactor this one so any changes to the underlying endpoint won't reflect the new state.

Reply:
But refactoring this to a bulk update, what would be considered an error then?